lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -553,6 +559,25 @@ def create_enrollment_questionnaire_response(record: REDCapRecord, patient_refer
 
     vaccine_item = create_vaccine_item(record["vaccine"], record['vaccine_year'], record['vaccine_month'], 'dont_know')
 
+    # Set the UW housing group
+    housing_group = None
+    if record.get('uw_housing_group_a') == '1':
+        housing_group = 'a'
+    elif record.get('uw_housing_group_b') == '1':
+        housing_group = 'b'
+    elif record.get('uw_housing_group_c') == '1':
+        housing_group = 'c'
+    elif record.get('uw_housing_group_d') == '1':
+        housing_group = 'd'
+    elif record.get('uw_housing_group_e') == '1':
+        housing_group = 'e'
+    elif record.get('uw_housing_group_f') == '1':
+        housing_group = 'f'
+    elif record.get('uw_housing_group_g') == '1':
+        housing_group = 'g'
+    record['uw_housing_group'] = housing_group
```

Comment:
Two questions about this: 
1. Do we care if more than one group equals "1" ? 
2. Do we care if `uw_housing_resident` is True, but none of these groups have calculated to "1"? 

These are QC things that would be nice to know about through warning messages?

Reply:
Ah, an interesting topic: QC, who does it and where is it done?
My take is not to bake much QC into the ETL job. It should warn about just the pieces that matter to it (an unexpected event name, for example). It should warn about things that the Dev team should drop everything and investigate.
For the housing questions, REDCap reports could find those problems. It's painful to run the reports because the project is big, but I don't think that justifies adding more QC to the Dev team.
This is a really good question for me and Kristen to ask "the BBI". I could see doing research QC as a valuable part of our workload.
I'm calling this out to @seattleflu/id3c-devs to raise this as something we should talk about.

Reply:
Yup, QC is always an interesting topic 😅 

I'm asking in this case because I don't have context on how these fields will be used downstream of the ETL. If these are important in the quota system or reporting to EHS, it would be good to know about these problems earlier.

Reply:
These fields will be important for test offering, so they are important. I agree that it's good to know about issues with these, but that could be done in a REDCap report instead of a warning in id3c-alerts. But this would require someone running the report once in a while to check.

Reply:
With this if/elif statement, we are giving "preferential" selection to group A. Could this accidentally inflate the count of people in group A that then messes up the priority queue? (Asking because I'm not sure how this variable will affect the priority queue)

Reply:
The calculation rules for the uw_housing_group_a through _g fields put a participant who lives in UW housing into one of the 7 housing groups based on the PT's REDCap record number. The record ranges for the 7 fields' logic don't overlap, so a PT should not be in more than one group.

The priority queue will invite the PTs for testing when it is their group's day ... for example, invite group "c" PTs on Wednesdays.

Reply:
Assuming there's only one possible housing group with a value of '1', a more concise way to write this logic would be:

```
groups = 'abcdefg'
housing_group = next( g for g in groups if record.get(f"housing_group_{group}") == '1')
```

Reply:
That's a really cool use of string's iterator! 
Not every participant will be in a uw_housing_group. It's only PTs who live on campus (apartments or dorms) that will get assigned to one of the uw_housing_groups.
When I try out this scenario--the PT having 0 for all the 7 housing groups--this throws an error when it gets to the end of the groups string (it throws on the iteration after g = 'g'). What we would want for this scenario is for housing_group to be None.
Is there a way to make that handle that case?

Reply:
Yep, you use `next()` in a try/except block and catch a `StopIteration` error to handle the special case.

Reply:
Gotcha. Because the majority of participants won't have a '1' for any of the uw_housing_group_ fields, we would hit the except block the majority of the time. Given that, I'm not going to go with the next() approach for this logic.